### PR TITLE
feat: add `array/base/mskreject-map`

### LINKED ISSUE #1323: [RFC]: add `@stdlib/array/base/mskreject-map`
### Description

This RFC proposes adding the package `@stdlib/array/base/mskreject-map`.

The package should be structured similarly to [`@stdlib/array/base/mskreject`](https://github.com/stdlib-js/stdlib/tree/64931c5fdb4abc3a339b619658d3b5acc089e556/lib/node_modules/%40stdlib/array/base/mskreject).

The package should expose an API having the same signature as `mskreject`, with the addition of `clbk` and `thisArg` arguments.

```javascript
function mskrejectMap( x, mask, clbk[, thisArg ] ) {...}
```

The callback function should map each selected value specified by `mask` to the output value.

In short, this package combines `mskreject` and `map` into a single API call.

Package: `@stdlib/array/base/mskreject-map`
Alias: `mskrejectMap`

### Related Issues

Related issues #1321, #1322 .

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by bad-in-coding**
Resolves #1323 .

## Description

> What is the purpose of this pull request?

This pull request adds the package `@stdlib/array/base/mskreject-map`.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #1323 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by Planeshifter** [comment]
@Ut-the-pro Would you mind looking into and fixing the lint failures first, please? (see failing CI checks)
Thanks!

**Comment by bad-in-coding** [comment]
@Planeshifter I am unable to figure out why it is showing lint error (in CI checks) for `Unexpected 'this'` in `test.assign.js`. Can you please suggest the change. Thanks.

**Comment by Planeshifter** [comment]
@Ut-the-pro If `this` is needed for the tests, you can disable the lint rule via either `// eslint-disable-line no-invalid-this` at the end of the offending line or `// eslint-disable-next-line no-invalid-this`.

**Comment by bad-in-coding** [comment]
@Planeshifter Can you please review it :)

File: lib/node_modules/@stdlib/array/base/mskreject-map/README.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+-->
+
+# mskreject-map
+
+> Apply a mask to a provided input array and map the unmasked values according to the clbk function provided.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/array/base/mskreject-map/README.md:23]:
```suggestion
> Apply a mask to a provided input array and map the unmasked values according to a mapping function.
```

File: lib/node_modules/@stdlib/array/base/mskreject-map/README.md [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+## Usage
+
+```javascript
+var mskrejectMap = require( '@stdlib/array/base/mskreject-map' );
+```
+
+#### mskrejectMap( x, mask, clbk\[, thisArg ] )
+
+Returns a new array by applying a mask and mapping the unmasked values according to the clbk function to the provided input array.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/array/base/mskreject-map/README.md:35]:
```suggestion
Returns a new array by applying a mask and mapping the unmasked values according to a mapping function.
```

File: lib/node_modules/@stdlib/array/base/mskreject-map/README.md [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+
+var out = mskrejectMap( x, mask, clbk, increase );
+// returns [ 4, 6 ]
+```
+
+The function **always** returns a new "generic" array.
+
+#### mskreject.assign( x, mask, out, stride, offset, clbk\[, thisArg ] )
+
+Applies a mask to a provided input array, maps the unmasked values according to the clbk function and assigns to elements in a provided output array.
```
Comment (Planeshifter) [inline lib/node_modules/@stdlib/array/base/mskreject-map/README.md:82]:
```suggestion
Applies a mask to a provided input array, maps the unmasked values according to a mapping function, and assigns to elements in a provided output array.
```

**Comment by bad-in-coding** [comment]
@Planeshifter I got a mail with some suggested changes in README but I don't see it here. Should I apply those changes???

**Review by Planeshifter — APPROVED** [review]
Let's get this in; thank you @Ut-the-pro!

**Pull request merged**